packages/core/core/src/Parcel.js

```diff
@@ -92,6 +95,41 @@ export default class Parcel {
 
     this.#runPackage = this.#farm.mkhandle('runPackage');
 
+    await this.initializeWatcher();
+  }
+
+  async initializeWatcher() {
+    let projectRoot = this.#resolvedOptions.projectRoot;
+    if (this.#resolvedOptions.watch) {
+      // TODO: ideally these should all be absolute paths already set up on #resolvedOptions
+      let targetDirs = this.#resolvedOptions.targets.map(target =>
+        path.resolve(process.cwd(), target.distDir)
+      );
+      let cacheDir = path.resolve(
+        process.cwd(),
+        this.#resolvedOptions.cacheDir
+      );
+      let vcsDirs = ['.git', '.hg'].map(dir => path.join(projectRoot, dir));
+      let ignore = [cacheDir, ...targetDirs, ...vcsDirs];
+      this.#watcher = await watcher.subscribe(
+        projectRoot,
+        (err, events) => {
+          if (err) {
+            throw err;
+          }
+
+          this.#assetGraphBuilder.respondToFSEvents(events);
+          if (this.#assetGraphBuilder.isInvalid()) {
+            this.build().catch(() => {
+              // Do nothing, in watch mode reporters should alert the user something is broken, which
+              // allows Parcel to gracefully continue once the user makes the correct changes
```

Comment:
maybe do the same `instanceof BuildError` check here?

Reply:
Wouldn't it only ever be an instance of `BuildError`?  Since `build()` catches all errors and either swallows them if their a `BuildAbortError` or wraps them in a `BuildError` and throws.